apps/storefront/lib/auth/SaleorAuthStorageHandler.ts

```diff
@@ -0,0 +1,60 @@
+/* auth state when user signs in / out */
+export const STORAGE_AUTH_EVENT_KEY = "saleor_storage_auth_change";
+export const STORAGE_AUTH_STATE_KEY = "saleor_auth_module_auth_state";
+
+const REFRESH_TOKEN_KEY = "saleor_auth_module_refresh_token";
+
+export type AuthState = "signedIn" | "signedOut";
+
+export type SaleorAuthEvent = CustomEvent<{ authState: AuthState }>;
+
+export class SaleorAuthStorageHandler {
```

Comment:
Isn't this the same logic we've added for checkout?
Maybe it should be extracted into a separate package in monorepo?